requirements-min.txt

```diff
@@ -1,6 +1,6 @@
 # minimum versions of package dependencies for installing PyNWB
-h5py==2.10  # support for selection of datasets with list of indices added in 2.10
+h5py==3.2.0
 hdmf==3.14.5
-numpy==1.18
-pandas==1.1.5
-python-dateutil==2.7.3
+numpy==1.23.0
```

Comment:
numpy 1.23.0 introduced updates to [structured (void) dtype comparisons](https://numpy.org/doc/stable/release/1.23.0-notes.html#changes-to-structured-void-dtype-promotion-and-comparisons) used to compare the scalars with compound data type in `ElectrodeGroup.position`. 

If 1.23.0 is too high of a minimum, we could go back to 1.20.0 and handle the comparisons ourselves in [`_assert_array_equal`](https://github.com/hdmf-dev/hdmf/blob/d1d1b5a654a4d0887929770f0d48493ab99621da/src/hdmf/testing/testcase.py#L160-L167) in hdmf

Reply:
I think this is OK. We can adjust if someone has a good reason to use numpy < 1.23.0